docs/en/reference/types.rst

```diff
@@ -909,3 +909,55 @@ hook it into the database platform:
 This would allow using a money type in the ORM for example and
 have Doctrine automatically convert it back and forth to the
 database.
+
+It is also possible to register type instances directly, in case you
+need to pass parameters to your instance::
+
+    <?php
+    namespace My\Project\Types;
+
+    use Doctrine\DBAL\Types\Type;
+    use Doctrine\DBAL\Platforms\AbstractPlatform;
+
+    final class EncryptedType extends StringType
+    {
+        public function __construct(
+            private string $cipher,
+            private string $secretName,
+            private SecretStore $secretStore
+        ) {
+        }
+
+        public function convertToPHPValue($value, AbstractPlatform $platform): string
+        {
+            return openssl_decrypt(
+                $value,
+                $this->cipher,
+                $this->secretStore->fetch($this->secretName)
+            );
+        }
+
+        public function convertToDatabaseValue($value, AbstractPlatform $platform): string
+        {
+            return openssl_encrypt(
+                $value,
+                $this->cipher,
+                $this->secretStore->fetch($this->secretName)
+            );
+        }
+
+        public function getName(): string
+        {
+            return 'encrypted';
+        }
+    }
+
+To do that, you can obtain the ``TypeRegistry`` singleton from ``Type``
+and register your type in it::
+
+    <?php
+    Type::getTypeRegistry()->addType('encrypted', new EncryptedType(
+        'aes-256-cbc-hmac-sha256',
+        '/path/to/secret',
+        $secretStore
+    ));
```

Comment:
Although this paragraph is a how-to, the one above it is as well, and the DBAL docs do not have a cookbook/how-to section. Not sure what to do.

Reply:
It's fine to keep it here.